### PR TITLE
LibJS: Implement Object.freeze(), Object.seal(), Object.isFrozen() and Object.isSealed()

### PR DESCRIPTION
**Pull request by linusg**
(not related to Dromaeo :smile:)

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/Object.cpp [hunk 1]
```diff
@@ -187,64 +187,61 @@ Value Object::get_own_property(const PropertyName& property_name, Value receiver
     return value_here;
 }
 
-Value Object::get_own_properties(PropertyKind kind, bool only_enumerable_properties, GetOwnPropertyReturnType return_type) const
+Vector<Value> Object::get_own_properties(PropertyKind kind, bool only_enumerable_properties, GetOwnPropertyReturnType return_type) const
 {
-    auto* properties_array = Array::create(global_object());
+    Vector<Value> properties;
```
Comment (awesomekling) [inline Userland/Libraries/LibJS/Runtime/Object.cpp:266]: These values are not protected from GC which may happen later in this function. This is why we have `MarkedValueList` :)

Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Object.cpp:266]: Hmm, but nothing was protecting the array (and therefore all the elements added to it) from being GC'd either, right?

Comment (awesomekling) [reply Userland/Libraries/LibJS/Runtime/Object.cpp:266]: A pointer to the array was on the stack, and our conservative GC scans everything on the stack that looks like a pointer. :) `Vector` storage is on the malloc heap which is not scanned by GC. See implementation in `Heap::gather_conservative_roots`.

Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Object.cpp]: > scans everything on the stack

that's the part I wasn't aware of :)
Updated!

**Pull request merged**